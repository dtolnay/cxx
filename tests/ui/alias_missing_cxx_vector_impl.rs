#[cxx::bridge]
mod here {
    struct Shared {
        z: usize,
    }

    extern "C" {
        type C;
    }
}

// Rustfmt mangles the extern type alias.
// https://github.com/rust-lang/rustfmt/issues/4159
#[rustfmt::skip]
#[cxx::bridge]
mod there {
    type Shared = crate::here::Shared;

    extern "C" {
        type C = crate::here::C;

        fn c_take_unique_ptr_vector(s: UniquePtr<CxxVector<C>>);
        fn c_take_unique_ptr_vector_shared(s: UniquePtr<CxxVector<Shared>>);
    }
}

fn main() {}
