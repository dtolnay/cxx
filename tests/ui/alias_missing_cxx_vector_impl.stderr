error[E0277]: the trait bound `here::C: VectorElement` is not satisfied
  --> $DIR/alias_missing_cxx_vector_impl.rs:15:1
   |
15 | #[cxx::bridge]
   | ^^^^^^^^^^^^^^ the trait `VectorElement` is not implemented for `here::C`
   |
  ::: $WORKSPACE/src/unique_ptr.rs
   |
   |     T: UniquePtrTarget,
   |        --------------- required by this bound in `UniquePtr`
   |
   = note: required because of the requirements on the impl of `UniquePtrTarget` for `CxxVector<here::C>`

error[E0277]: the trait bound `here::Shared: VectorElement` is not satisfied
  --> $DIR/alias_missing_cxx_vector_impl.rs:15:1
   |
15 | #[cxx::bridge]
   | ^^^^^^^^^^^^^^ the trait `VectorElement` is not implemented for `here::Shared`
   |
  ::: $WORKSPACE/src/unique_ptr.rs
   |
   |     T: UniquePtrTarget,
   |        --------------- required by this bound in `UniquePtr`
   |
   = note: required because of the requirements on the impl of `UniquePtrTarget` for `CxxVector<here::Shared>`
